converter/solid_figures.py

```diff
@@ -0,0 +1,79 @@
+from dataclasses import dataclass
+from abc import ABC
+
+
+@dataclass(frozen=True)
+class SolidFigure(ABC):
+    """
+    Abstract solid figure in 3D space. It is characterised by position in
+    space and rotation along X,Y,Z axis in its own reference frame. Size of
+    the figure (its extend in space) is defined in its subclasses.
+    """
+
+    uuid: str
+    offset: tuple[float, float, float]
```

Comment:
```suggestion
    position: tuple[float, float, float]
```

`Offset` is too much computer-oriented. In the language of solids in 3D space we don't speak much about offset.
Position is a better name. In more details: position of a reference point (i.e. solid center).

Reply:
Offset it is then 😄